src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = "white";
+    private static final double DEFAULT_RADIUS = 10.0;
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public AbstractFigure getRandomFigure() {
+        Random random = new Random();
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                double squareSide = getRandomParameter();
+                return new Square(color, squareSide);
+            case 1:
+                double rectangleLength = getRandomParameter();
+                double rectangleWidth = getRandomParameter();
+                return new Rectangle(color, rectangleLength, rectangleWidth);
+            case 2:
+                double triangleFirstLeg = getRandomParameter();
+                double triangleSecondLeg = getRandomParameter();
+                return new RightTriangle(color, triangleFirstLeg, triangleSecondLeg);
+            case 3:
+                double circleRadius = getRandomParameter();
+                return new Circle(color, circleRadius);
+            default:
+                double trapezoidUpperBase = random.nextDouble() * 10 + 1;
+                double trapezoidLowerBase = random.nextDouble() * 10 + 1;
+                double trapezoidHeight = random.nextDouble() * 10 + 1;
+                return new IsoscelesTrapezoid(color,
+                        trapezoidUpperBase,
+                        trapezoidLowerBase,
+                        trapezoidHeight);
+        }
```

Comment:
As Anastasia asked before it's better to create separate method for each figure creation and move logic there.
```suggestion
            case 0:
                return getRandomSquare();
            case 1:
                return getRandomRectangle();
            case 2:
                return getRandomRightTriangle();
            case 3:
                return getRandomCircle();
            default:
                return getRandomIsoscelesTrapezoid();
        }
```
It will increase code readability, especially for public method